pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -4082,6 +4092,10 @@ public PositionImpl getMaxReadPosition() {
         return this.transactionBuffer.getMaxReadPosition();
     }
 
+    public CompletableFuture<Position> getLastCanDispatchPositionWithTxn() {
```

Comment:
Maybe we can rename the method to `getLastDispatchablePositionWithTxn` or something else.

Reply:
Make sense to me.